src/get-started/flutter-for/ios-devs.md

```diff
@@ -966,264 +1584,379 @@ class _SampleAppPageState extends State<SampleAppPage> {
+  @override
+  void paint(Canvas canvas, Size size) {
+    final Paint paint = Paint()
+      ..color = Colors.black
+      ..strokeCap = StrokeCap.round
+      ..strokeWidth = 5.0;
+    for (int i = 0; i < points.length - 1; i++) {
+      if (points[i] != null && points[i + 1] != null) {
+        canvas.drawLine(points[i]!, points[i + 1]!, paint);
+      }
+    }
+  }
 
-<?code-excerpt "lib/images.dart (AssetImage)"?>
-```dart
-AssetImage('images/a_dot_burr.jpeg')
+  @override
+  bool shouldRepaint(SignaturePainter oldDelegate) =>
+      oldDelegate.points != points;
+}
 ```
 
-or directly in an `Image` widget:
+### Widget opacity
 
-<?code-excerpt "lib/images.dart (Imageasset)"?>
-```dart
-@override
-Widget build(BuildContext context) {
-  return Image.asset('images/my_image.png');
-}
-```
+In UIKit, everything has `.opacity` or `.alpha`.
+In Flutter, most of the time you need to
+wrap a widget in an `Opacity` widget to accomplish this.
 
-For more details, see
-[Adding Assets and Images in Flutter][].
+### Custom Widgets
 
-### Where do I store strings? How do I handle localization?
+In UIKit, you typically subclass `UIView`, or use a pre-existing view,
+to override and implement methods that achieve the desired behavior.
+In Flutter, build a custom widget by [composing][] smaller widgets
+(instead of extending them).
 
-Unlike iOS, which has the `Localizable.strings` file,
-Flutter doesn't currently have a dedicated system for handling strings.
-At the moment, the best practice is to declare your copy text
-in a class as static fields and access them from there. For example:
+For example, how do you build a `CustomButton`
+that takes a label in the constructor?
+Create a CustomButton that composes a `ElevatedButton` with a label,
+rather than by extending `ElevatedButton`:
 
-<?code-excerpt "lib/string_examples.dart (Strings)"?>
+<?code-excerpt "lib/custom.dart (CustomButton)"?>
 ```dart
-class Strings {
-  static const String welcomeMessage = 'Welcome To Flutter';
+class CustomButton extends StatelessWidget {
+  const CustomButton(this.label, {super.key});
+
+  final String label;
+
+  @override
+  Widget build(BuildContext context) {
+    return ElevatedButton(
+      onPressed: () {},
+      child: Text(label),
+    );
+  }
 }
 ```
 
-You can access your strings as such:
+Then use `CustomButton`,
+just as you'd use any other Flutter widget:
 
-<?code-excerpt "lib/string_examples.dart (AccessString)" replace="/const //g"?>
+<?code-excerpt "lib/custom.dart (UseCustomButton)"?>
 ```dart
-Text(Strings.welcomeMessage);
+@override
+Widget build(BuildContext context) {
+  return const Center(
+    child: CustomButton('Hello'),
+  );
+}
 ```
 
-By default, Flutter only supports US English for its strings.
-If you need to add support for other languages,
-include the `flutter_localizations` package.
-You might also need to add Dart's [`intl`][]
-package to use i10n machinery, such as date/time formatting.
+## Navigation
```

Comment:
Please add at least one brief line of text between a level 2 and a level 3 header. (Between any headers, to be honest.)